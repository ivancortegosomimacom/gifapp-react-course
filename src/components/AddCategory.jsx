import PropTypes from "prop-types";
import { useState } from "react";

export const AddCategory = ({ setCategories, onAddCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    const inpValProcessed = inputValue.trim();
    if (inpValProcessed.length > 1) {
      onAddCategory(inpValProcessed);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={onSubmit} aria-label="form-category">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired,
};
