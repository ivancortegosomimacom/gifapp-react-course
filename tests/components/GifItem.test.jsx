import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("GifItem tests", () => {
  const title = "Jujutsu Kaisen";
  const url = "https://url.test.jpg/";
  const titleError = 123;
  const urlError = 321;
  test("should match snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should show image with URL and ALT sent", () => {
    render(<GifItem title={title} url={url} />);
    // expect(screen.getByRole('img').src).toBe(url);
    // expect(screen.getByRole('img').alt).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });

  test("should show title inside component", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByAltText(title)).toBeTruthy();
  });
});
