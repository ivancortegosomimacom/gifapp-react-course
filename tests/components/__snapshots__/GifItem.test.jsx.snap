// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem tests should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Jujutsu Kaisen"
      src="https://url.test.jpg/"
    />
  </div>
</div>
`;
